### PR TITLE
Startup react

### PR DESCRIPTION
**Pull request by Octopus773**
Some said it couldn't be made
but here it is
a front end PR

### CODE REVIEW

File: web-app/src/App.tsx [hunk 1]
```diff
@@ -0,0 +1,128 @@
+import type { PipelineBoxProps } from "./components/Pipelines/PipelineBox";
+import { GenericButtonProps } from "./components/GenericButton";
+import { Typography, Box, Button } from "@mui/material";
+import type { ImageProps } from "./components/types";
+import "./App.css";
+
+
+import { useNavigate } from "react-router-dom";
+import MoreVertIcon from "@mui/icons-material/MoreVert";
+import * as React from "react";
```
Comment (GitBluub) [inline web-app/src/App.tsx:10]: is it necessary to import everything from the react module ?

File: web-app/src/components/Pipelines/GameBox.fetsx [hunk 2]
```diff
@@ -0,0 +1,199 @@
+import React from 'react';
```
Comment (GitBluub) [inline web-app/src/components/Pipelines/GameBox.fetsx:1]: What is this gamebox file, ends in "fetsx" too ?

File: web-app/src/components/Pipelines/PipelineCreation.tsx [hunk 3]
```diff
@@ -0,0 +1,110 @@
+import StepButton from "@mui/material/StepButton";
+import Typography from "@mui/material/Typography";
+import Stepper from "@mui/material/Stepper";
+import Button from "@mui/material/Button";
+import Step from "@mui/material/Step";
+import Box from "@mui/material/Box";
+import * as React from "react";
```
Comment (GitBluub) [inline web-app/src/components/Pipelines/PipelineCreation.tsx:7]: Avoid importing all of the module, it shouldn t be as needed

File: web-app/src/components/Pipelines/PipelineCreation.tsx [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+
+const steps = ["Select campaign settings", "Create an ad group", "Create an ad"];
```
Comment (GitBluub) [inline web-app/src/components/Pipelines/PipelineCreation.tsx:9]: lets create an ad

File: web-app/src/components/Pipelines/PipelineModal.tsx [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+interface PipelineModalProps {
+	isOpen: boolean;
+	children: React.ReactNode;
+	handleClose: () => void;
+}
+
+export default function PipelineModal({ isOpen, children, handleClose }: PipelineModalProps) {
+	const classes = useStyles();
+	const theme = useTheme();
+	//             <Button onClick={handleOpen}>Show pipeline</Button>
```
Comment (GitBluub) [inline web-app/src/components/Pipelines/PipelineModal.tsx:34]: useless comment, you won t put this button anyway

File: web-app/src/index.tsx [hunk 6]
```diff
@@ -0,0 +1,45 @@
+import * as React from "react";
```
Comment (GitBluub) [inline web-app/src/index.tsx:1]: import everything

File: web-app/src/index.tsx [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+	<React.StrictMode>
+		<AerisRouter />
+	</React.StrictMode>,
+	document.getElementById("root")
+);
+
+// If you want to start measuring performance in your app, pass a function
+// to log results (for example: reportWebVitals(console.log))
+// or send to an analytics endpoint. Learn more: https://bit.ly/CRA-vitals
+reportWebVitals();
```
Comment (GitBluub) [inline web-app/src/index.tsx:45]: remove this if you d idn t setup web vitals

File: web-app/src/pages/HomePage.tsx [hunk 8]
```diff
@@ -0,0 +1,191 @@
...
+import ServiceSetupModal from "./ServiceSetup";
+import { AppServices, ServiceActions, AppServicesLogos } from "../utils/globals";
+
+const useStyles = makeStyles((theme) => ({
+	divHomePage: {
+		display: "contents",
+	},
+}));
+
+const randomProperty = function (obj: Object): any {
```
Comment (GitBluub) [inline web-app/src/pages/HomePage.tsx:26]: please remove this :vomiting_face:

File: web-app/src/pages/HomePage.tsx [hunk 9]
```diff
@@ -0,0 +1,191 @@
...
+					"& > :not(style)": { m: 1 },
+					position: "fixed",
+					bottom: "5px",
+					right: "5px",
+				}}>
+				<Fab
+					size="medium"
+					color="secondary"
+					aria-label="add"
+					onClick={() => {
```
Comment (GitBluub) [inline web-app/src/pages/HomePage.tsx:171]: put this onClick on a function above for code clarity

File: web-app/src/pages/Login/LoginPage.tsx [hunk 10]
```diff
@@ -0,0 +1,335 @@
+import React, { useReducer, useEffect } from "react";
```
Comment (GitBluub) [inline web-app/src/pages/Login/LoginPage.tsx:1]: can t review this

File: web-app/package.json [hunk 11]
```diff
@@ -0,0 +1,62 @@
...
+	"version": "0.1.0",
+	"private": true,
+	"dependencies": {
+		"@emotion/react": "^11.7.1",
+		"@emotion/styled": "^11.6.0",
+		"@mui-treasury/components": "^1.10.1",
+		"@mui/icons-material": "^5.3.1",
+		"@mui/lab": "^5.0.0-alpha.68",
+		"@mui/material": "^5.3.1",
+		"@testing-library/jest-dom": "^5.16.1",
```
Comment (AnonymusRaccoon) [inline web-app/package.json:12]: I don't think testing-libraries are dependencies, same for @types, that should be in dev-devencies.

File: web-app/package.json [hunk 12]
```diff
@@ -0,0 +1,62 @@
...
+		"react": "^17.0.2",
+		"react-dom": "^17.0.2",
+		"react-router-dom": "^6.2.1",
+		"react-scripts": "5.0.0",
+		"typescript": "^4.5.4",
+		"web-vitals": "^2.1.3"
+	},
+	"scripts": {
+		"start": "react-scripts start",
+		"webpack": "webpack",
```
Comment (AnonymusRaccoon) [inline web-app/package.json:30]: If you don't use webpack, remove it.

File: web-app/src/components/Pipelines/PipelineModal.tsx [hunk 13]
```diff
@@ -0,0 +1,67 @@
...
+
+	return (
+		<div>
+			<Modal
+				className={classes.modal}
+				aria-labelledby="simple-modal-title"
```
Comment (AnonymusRaccoon) [inline web-app/src/components/Pipelines/PipelineModal.tsx:40]: Is this aria really the best title possible?

File: web-app/src/pages/HomePage.tsx (hunk 8 above)
Comment (Octopus773) [reply web-app/src/pages/HomePage.tsx:26]: lol

**Review by Arthi-chaud — APPROVED** [review]
to long to care

**Review by 0Nom4D — APPROVED** [review]

**Review by GitBluub — APPROVED** [review]

File: web-app/src/pages/ServiceSetup.tsx [hunk 14]
```diff
@@ -0,0 +1,71 @@
+import Typography from "@mui/material/Typography";
+import { Login, Logout } from "@mui/icons-material";
+import Box from "@mui/material/Box";
+
+import GenericButton from "../components/GenericButton";
+import { ImageProps } from "../components/types";
+import { Grid } from "@mui/material";
+import { AppServices } from "../utils/globals";
+
+export default function ServiceSetupModal() {
```
[the remark this answers is not in the document]
Comment (AnonymusRaccoon) [reply web-app/src/pages/ServiceSetup.tsx:10]: You are 40 years old.

**Review by AnonymusRaccoon — APPROVED** [review]

**Pull request merged**